app/helpers/results_helper.rb

```diff
@@ -1,5 +1,36 @@
 module ResultsHelper
-  def player_options
-    Player.order("name ASC").all.map { |player| [player.name, player.id] }
+  def player_options(index, current_player)
+    current = [[current_player.name, current_player.id]]
+    all = Player.order("name ASC").all.map { |p| [p.name, p.id] }
+    if index == 0
+      current
+    else
+      all - current
+    end
+  end
+
+  def relation_options
```

Comment:
In the context of table tennis we don't need `ties` anyway. So I didn't include the `allow_ties` argument.